src/java/apiview-java-processor/src/main/java/com/azure/tools/apiview/processor/analysers/JavaASTAnalyser.java

```diff
@@ -190,8 +192,13 @@ public class JavaASTAnalyser implements Analyser {
      **********************************************************************************************/
 
     public JavaASTAnalyser(APIListing apiListing) {
+        this(apiListing, null, true);
+    }
+
+    public JavaASTAnalyser(APIListing apiListing, SymbolCollector collector, boolean enableDiagnostics) {
         this.apiListing = apiListing;
-        this.diagnostic = new Diagnostics(apiListing);
+        this.diagnostic = new Diagnostics(apiListing, enableDiagnostics);
```

Comment:
If it were me, I would define `Diagnostics` as an interface, and then have the default implementation called something like `AzureSdkDiagnostics`. Then, I would change the `JavaASTAnalyser` to accept a `Diagnostics` parameter. By default you would then create an `AzureSdkDiagnostics` instance in main and pass it in, and in the diff mode you would create an abstract instance, e.g.

```
Diagnostics noOpDiagnostics = new Diagnostics() { 
    public void scanFinal(...)  { }
    public void scanIndividual(..) { }
};
JavaASTAnalyser analyser = new JavaASTAnalyser(apiListing, noOpDiagnostics);
```